src/transport_ofi.h

```diff
@@ -539,19 +555,36 @@ static inline
 void
 shmem_transport_get_wait(void)
 {
-  SHMEM_MUTEX_LOCK(shmem_transport_ofi_lock);
-	int ret = 0;
+    SHMEM_MUTEX_LOCK(shmem_transport_ofi_lock);
+
+    /* wait for get counter to meet outstanding count value */
+#ifdef ENABLE_COMPLETION_POLLING
+    uint64_t success, fail;
+    do {
+        success = fi_cntr_read(shmem_transport_ofi_get_cntrfd);
+        fail = fi_cntr_readerr(shmem_transport_ofi_get_cntrfd);
 
-	/* wait for get counter to meet outstanding count value    */
-	ret = fi_cntr_wait(shmem_transport_ofi_get_cntrfd,
-			shmem_transport_ofi_pending_get_counter,-1);
-    if(ret) {
-	    struct fi_cq_err_entry e = {0};
-        fi_cq_readerr(shmem_transport_ofi_put_nb_cqfd,
-		           (void *)&e, 0);
+        if (success < shmem_transport_ofi_pending_put_counter && fail == 0) {
```

Comment:
I don't think there is a mixup at the line you flagged, but I did have a mixup in get_wait().  Pushed a patch to fix this, although, I'm a little concerned that it didn't show up in testing...